File: backend/src/main/scala/microservices/content/objects/ExploreSearchResultPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ExploreSearchResultPlannerResponse(
    resourceType: String,
    resourceId: String,
    title: String,
    summary: String,
    metaLabel: String,
    navigationHint: String,
    imageUrl: Option[String],
    score: Int
)
object ExploreSearchResultPlannerResponse:
  given sourceEncoder: Encoder[ExploreSearchResultPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ExploreSearchResultPlannerResponse] = deriveDecoder

File: frontend/src/microservices/content/objects/ExploreSearchResultPlannerResponse.ts
// ExploreSearchResultPlannerResponse：content 域探索搜索单条结果对象。

export type ExploreSearchResultPlannerResponse = {
  resourceType: string
  resourceId: string
  title: string
  summary: string
  metaLabel: string
  navigationHint: string
  imageUrl: string | null
}
export const exploreSearchResultPlannerResponseFromJson = (json: string): ExploreSearchResultPlannerResponse =>
  JSON.parse(json) as ExploreSearchResultPlannerResponse
export const exploreSearchResultPlannerResponseToJson = (value: ExploreSearchResultPlannerResponse): string =>
  JSON.stringify(value)
